clientapi/routing/membership.go

```diff
@@ -128,17 +128,16 @@ func SendMembership(
 			RoomID string `json:"room_id"`
 		}{roomID}
 	default:
-	}
-
-	_, err = producer.SendEvents(
-		req.Context(),
-		[]gomatrixserverlib.HeaderedEvent{event.Headered(verRes.RoomVersion)},
-		cfg.Matrix.ServerName,
-		nil,
-	)
-	if err != nil {
-		util.GetLogger(req.Context()).WithError(err).Error("producer.SendEvents failed")
-		return jsonerror.InternalServerError()
+		_, err = producer.SendEvents(
```

Comment:
You're missing the `fallthrough` you had before for `case gomatrixserverlib.Join`

Reply:
Ah whoops, well spotted!